sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -137,13 +139,98 @@ public AnalyzedRelation analyze(Node node, Analysis analysis) {
     }
 
     @Override
-    protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext context) {
-        return process(node.getQueryBody(), context);
+    protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext statementContext) {
+        // In case of Set Operation or VALUES clause
```

Comment:
Maybe expand this comment. Special cases like this can be confusing to the reader and should be documented well.